### ISSUE TITLE
[Help Wanted]: Gopls LSP settings not being updated

### ISSUE THREAD

**Issue by manofthelionarmy** [ChristianChiarulli/LunarVim#624]
### Problem description

I was hoping to change the hoverKind for the hover doc for gopls with hopes the method signature is included for functions. No changes are taking place :( 

Anything I'm missing or not understanding?  (most likely something really simple I'm missing) 

1. :LspInstall go
2. nvim ~/.config/ftplugin/go.lua
3. Change hoverKind to SingleLine, as such:
```
require'lspconfig'.gopls.setup {
    cmd = {DATA_PATH .. "/lspinstall/go/gopls"},
    settings = {gopls = {analyses = {unusedparams = true}, staticcheck = true}, hoverKind = "SingleLine"},
    root_dir = require'lspconfig'.util.root_pattern(".git", "go.mod"),
    init_options = {usePlaceholders = true, completeUnimported = true },
    on_attach = require'lsp'.common_on_attach
}

if O.lang.go.autoformat then
    require('lv-utils').define_augroups({
        _go_format = {
            {'BufWritePre', '*.go', 'lua vim.lsp.buf.formatting_sync(nil,1000)'}
        },
        _go = {
            -- Go generally requires Tabs instead of spaces.
            {'FileType', 'go', 'setlocal tabstop=4'},
            {'FileType', 'go', 'setlocal shiftwidth=4'},
            {'FileType', 'go', 'setlocal softtabstop=4'},
            {'FileType', 'go', 'setlocal noexpandtab'}
        }
    })
end

```

### LunarVim version

stable

### Neovim version (>= 0.5)

0.5

### Relevant log output

```shell
NVIM v0.5.0
Build type: Release
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/cc -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -O2 -DNDEBUG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/armando/neovim/build/config -I/home/armando/neovim/src -I/home/armando/neovim/.deps/usr/include -I/usr/include -I/home/armando/neovim/build/src/nvim/auto -I/home/armando/neovim/build/include
Compiled by root@armando-Q524UQK

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/share/nvim"

Run :checkhealth for more info
```

### Screenshots

Expected behavior: 

![Screenshot from 2021-07-03 11-41-13](https://user-images.githubusercontent.com/11958828/124364073-fb9f7780-dbf3-11eb-9579-8e7198884054.png)

Actual Behavior: 

![Screenshot from 2021-07-03 11-19-15](https://user-images.githubusercontent.com/11958828/124363578-e70db000-dbf0-11eb-9b0b-b111ceb1293c.png)

### I have read

- [X] The readme
- [X] The wiki

**Comment by manofthelionarmy** [comment]
I see I can accomplish what I see by typing `Esc` and then "shift-k", followed by typing 'shift-a' and the hover_doc will stay up. I guess this will take some practice so I can get use to this new flow :)

**Comment by manofthelionarmy** [comment]
Okay I'll close this out, this is a totally a misunderstanding, on my end, of how nvim lsp works regarding signature hints. Signature hints can be done with `Shift-K` while in command mode. I'll update the name of this so anyone who wishes to accomplish the same can find it. I was hoping to get signature hints while in insert mode, but I found a package that does this :) 

https://github.com/ray-x/lsp_signature.nvim

I added it to `~/.config/nvim/lua/plugins.lua` as stated in the readme of the above repo I shared. 

Lastly, I added this to `common_on_attach` in `~/.config/nvim/lua/lsp/init.lua`, starting at line 102:

```
if O.document_highlight then
    function lsp_config.common_on_attach(client, bufnr)
        require "lsp_signature".on_attach()  -- <- add it right here
        documentHighlight(client, bufnr)
    end
end

```
After doing that, I get signature hints while in insertion mode:

![Screenshot from 2021-07-04 12-16-59](https://user-images.githubusercontent.com/11958828/124396977-c9f4e200-dcc1-11eb-830b-96a288d99174.png)
